### PR TITLE
Modules rewrite, storage refactor, huge clenup

### PR DESCRIPTION
**Pull request by filrak**
Continuation of https://github.com/DivanteLtd/vue-storefront/issues/3144

### Short description and why it's useful

- Rewrite of old modules to new API by Bartosz Żądłowski (not fully ready yet)
- Unification of cache keys (some of theme were different than cache entities in StorageManager or config some had different setters than getters and worked only because of `get` fallback. Now cache/module/StorageManager/config keys are identical for all of the entities which should end up in easier debugging ;):
  - `syncTaskCollection` renamed to `syncTasks`
  - `compareCollection` renamed to `compare`
  - `cmsData` renamed to cms
  - `cartsCollection` renamed to carts
  - `checkoutFieldValues`, `checkoutFieldsCollection` renamed to `checkout` (there was different getter and setter for same entity, one was fully redundant)
  - ...much more (will be in upgrade notes)
- Improvements in StorageManager API (added 'init' function)
- renamed `initCacheStorage.ts` to `cache.ts`, add webpack alias for backward compatibility

More code was deleted than added so it's a good one ^^

TODO in the next iteration
- move storage manager to `core/lib` and rename to `CacheManager`

### CODE REVIEW

**Comment by pkarw** [comment]
@filrak please remember to add the upgrade notes with these collection name changes

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added configuration for max attempt task & cart by pass - @cnviradiya (#3193)
 - Added catching of errors when ES is down - @qiqqq
 - Added debounce for updating quantity method in the cart - @andrzejewsky (#3191)
-
+- New modules API and rewrite - @filrak (#3144)
```
Comment (patzick) [inline CHANGELOG.md]: please add @JCown input in changelog

File: core/store/lib/storage-manager.ts [hunk 2]
```diff
@@ -1,14 +1,28 @@
...
   /**
-   * Register the cache storage - this is required prior to accessing the collection
-   * @param collectionName string name of the cache collection to register
-   * @param collectionInstance UniversalStorage driver
+   * Register the cache storage index that can be later accessed and modified - this is required prior to accessing the collection
+   * @param collectionName name of the cache collection to create
+   * @param isLocalized if set to `false` data will be shared between storeViews (default `true`)
+   * @param storageQuota max size of storage, 0 if unlimited (default `0`)
+   */
+  init: function (collectionName: string, isLocalised = true, storageQuota = 0) {
```
Comment (patzick) [inline core/store/lib/storage-manager.ts]: as jsDoc param
```suggestion
  init: function (collectionName: string, isLocalized = true, storageQuota = 0) {
```

File: core/store/lib/storage-manager.ts [hunk 3]
```diff
@@ -1,14 +1,28 @@
...
+    this.storageMap[collectionName] = _prepareCacheStorage(collectionName, isLocalised, storageQuota)
```
Comment (patzick) [inline core/store/lib/storage-manager.ts]:
```suggestion
    this.storageMap[collectionName] = _prepareCacheStorage(collectionName, isLocalized, storageQuota)
```

File: core/store/lib/storage-manager.ts [hunk 4]
```diff
@@ -20,16 +34,47 @@ const StorageManager = {
...
+      Logger.warn('Called cache collection ' + collectionName + ' does not exist. Initializing.', 'cache')
+      return this.set(collectionName, initCacheStorage(collectionName, true))
     } else {
       return this.storageMap[collectionName]
     }
   }
 }
 
-export { StorageManager }
+function _prepareCacheStorage (key, localised = true, storageQuota = 0) {
```
Comment (patzick) [inline core/store/lib/storage-manager.ts]:
```suggestion
function _prepareCacheStorage (key, localized = true, storageQuota = 0) {
```

File: core/store/lib/storage-manager.ts [hunk 5]
```diff
@@ -20,16 +34,47 @@ const StorageManager = {
...
+  const storeView = currentStoreView()
+  const dbNamePrefix = storeView && storeView.storeCode ? storeView.storeCode + '-' : ''
+  const cacheDriver = config.localForage && config.localForage.defaultDrivers[key]
+    ? config.localForage.defaultDrivers[key]
+    : 'LOCALSTORAGE'
+
+  return new UniversalStorage(localForage.createInstance({
+    name: localised ? `${dbNamePrefix}shop` : 'shop',
```
Comment (patzick) [inline core/store/lib/storage-manager.ts]:
```suggestion
    name: localized ? `${dbNamePrefix}shop` : 'shop',
```

File: src/modules/payment-backend-methods/index.ts [hunk 6]
```diff
@@ -14,9 +15,34 @@ const store = {
...
-})
+export const PaymentBackendMethodsModule: StorefrontModule = function (app, store, router, moduleConfig, appConfig) {
+  store.registerModule('payment-backend-methods', PaymentBackendMethodsStore)
+
+  let correctPaymentMethod = false
+
+  // Place the order. Payload is empty as we don't have any specific info to add for this payment method '{}'
+  const placeOrder = () => {
+    if (correctPaymentMethod) {
+      Vue.prototype.$bus.$emit('checkout-do-placeOrder', {})
```
Comment (patzick) [inline src/modules/payment-backend-methods/index.ts]: shouldn't we stop using Vue.prototype.$bus? @pkarw did refactor with that and i think just EventBus should be used here

Comment (filrak) [reply src/modules/payment-backend-methods/index.ts]: this part was only moved but you're right ;)

**Comment by pkarw** [comment]
Great job!

**Pull request merged**